src/GitVersionExe/GitVersionExecutor.cs

```diff
@@ -10,24 +10,23 @@
 
 namespace GitVersion
 {
-    public class GitVersionApplication : IGitVersionApplication
+    public class GitVersionExecutor : IGitVersionExecutor
     {
         private readonly IFileSystem fileSystem;
-        private readonly IEnvironment environment;
         private readonly ILog log;
         private readonly IConfigFileLocator configFileLocator;
         private readonly IHelpWriter helpWriter;
+        private readonly IExecCommand execCommand;
         private readonly IVersionWriter versionWriter;
 
-        public GitVersionApplication(IFileSystem fileSystem, IEnvironment environment, ILog log, IConfigFileLocator configFileLocator)
+        public GitVersionExecutor(IFileSystem fileSystem, ILog log, IConfigFileLocator configFileLocator, IVersionWriter versionWriter, IHelpWriter helpWriter, IExecCommand execCommand)
         {
             this.fileSystem = fileSystem;
-            this.environment = environment;
             this.log = log;
             this.configFileLocator = configFileLocator;
-
-            versionWriter = new VersionWriter();
-            helpWriter = new HelpWriter(versionWriter);
+            this.versionWriter = versionWriter;
+            this.helpWriter = helpWriter;
+            this.execCommand = execCommand;
         }
 
         public int Run(Arguments arguments)
```

Comment:
With the class being named `Executor`, perhaps `Execute` is a more fitting method name than `Run`?

Reply:
Changed to `Execute`